<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>GLControlWidget</class>
 <widget class="QWidget" name="GLControlWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>677</width>
    <height>501</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QHBoxLayout" name="horizontalLayout_8">
   <item>
    <widget class="GLMeshPicture" name="picture" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>0</verstretch>
      </sizepolicy>
     </property>
    </widget>
   </item>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout_7">
     <property name="spacing">
      <number>0</number>
     </property>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_5">
       <item>
        <layout class="QVBoxLayout" name="verticalLayout_4">
         <property name="spacing">
          <number>0</number>
         </property>
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>Вращение (градусы):</string>
           </property>
           <property name="alignment">
            <set>Qt::AlignCenter</set>
           </property>
          </widget>
         </item>
         <item>
          <layout class="QHBoxLayout" name="horizontalLayout">
           <property name="spacing">
            <number>2</number>
           </property>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="labelX">
               <property name="text">
                <string>x</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="sliderX">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinX">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_2">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="labelY">
               <property name="text">
                <string>y</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="sliderY">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinY">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item>
            <layout class="QVBoxLayout" name="verticalLayout_3">
             <property name="spacing">
              <number>0</number>
             </property>
             <item>
              <widget class="QLabel" name="labelZ">
               <property name="text">
                <string>z</string>
               </property>
               <property name="alignment">
                <set>Qt::AlignCenter</set>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSlider" name="sliderZ">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
               <property name="orientation">
                <enum>Qt::Vertical</enum>
               </property>
               <property name="tickPosition">
                <enum>QSlider::TicksBothSides</enum>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QSpinBox" name="spinZ">
               <property name="sizePolicy">
                <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
                 <horstretch>0</horstretch>
                 <verstretch>0</verstretch>
                </sizepolicy>
               </property>
               <property name="maximum">
                <number>359</number>
               </property>
              </widget>
             </item>
            </layout>
           </item>
          </layout>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="text">
            <string>Зум:</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QDoubleSpinBox" name="spinZoom">
           <property name="sizePolicy">
            <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
             <horstretch>0</horstretch>
             <verstretch>0</verstretch>
            </sizepolicy>
           </property>
           <property name="minimum">
            <double>0.010000000000000</double>
           </property>
           <property name="maximum">
            <double>100.000000000000000</double>
           </property>
           <property name="singleStep">
            <double>0.010000000000000</double>
           </property>
           <property name="value">
            <double>1.000000000000000</double>
           </property>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_6">
       <property name="spacing">
        <number>0</number>
       </property>
       <item>
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>Манипулятор-мышь:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QComboBox" name="mouseMode">
         <item>
          <property name="text">
           <string>Вращение</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Перемещение</string>
          </property>
         </item>
         <item>
          <property name="text">
           <string>Выбор</string>
          </property>
         </item>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_3">
       <item>
        <widget class="QLabel" name="labelTranslateStep">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="text">
          <string>Шаг:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="translateStep">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="decimals">
          <number>7</number>
         </property>
         <property name="minimum">
          <double>0.000001000000000</double>
         </property>
         <property name="maximum">
          <double>32000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.000200000000000</double>
         </property>
         <property name="value">
          <double>0.010000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_5">
       <item>
        <widget class="QLabel" name="labelXTranslation">
         <property name="text">
          <string>Смещение x:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="xTranslation">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-100000.000000000000000</double>
         </property>
         <property name="maximum">
          <double>100000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_6">
       <item>
        <widget class="QLabel" name="labelYTranslation">
         <property name="text">
          <string>Смещение y:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="yTranslation">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-100000.000000000000000</double>
         </property>
         <property name="maximum">
          <double>100000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <layout class="QHBoxLayout" name="horizontalLayout_7">
       <item>
        <widget class="QLabel" name="labelZTranslation">
         <property name="text">
          <string>Смещение z:</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QDoubleSpinBox" name="zTranslation">
         <property name="decimals">
          <number>4</number>
         </property>
         <property name="minimum">
          <double>-100000.000000000000000</double>
         </property>
         <property name="maximum">
          <double>100000.000000000000000</double>
         </property>
         <property name="singleStep">
          <double>0.100000000000000</double>
         </property>
        </widget>
       </item>
      </layout>
     </item>
     <item>
      <widget class="Line" name="line_2">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QVBoxLayout" name="verticalLayout_8">
       <item>
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>Визуализация:</string>
         </property>
        </widget>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QComboBox" name="visualizationMode">
           <item>
            <property name="text">
             <string>цвет</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>узел</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>элемент</string>
            </property>
           </item>
          </widget>
         </item>
         <item>
          <widget class="QComboBox" name="colorMap">
           <item>
            <property name="text">
             <string>зима</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>весна</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>лето</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>осень</string>
            </property>
           </item>
           <item>
            <property name="text">
             <string>серость</string>
            </property>
           </item>
          </widget>
         </item>
        </layout>
       </item>
      </layout>
     </item>
     <item>
      <widget class="QCheckBox" name="showMesh">
       <property name="text">
        <string>Сетка</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QCheckBox" name="showRegionBorder">
       <property name="text">
        <string>Границы области</string>
       </property>
       <property name="checked">
        <bool>true</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>GLMeshPicture</class>
   <extends>QWidget</extends>
   <header>glmeshpicture.h</header>
   <container>1</container>
   <slots>
    <signal>xRotationChanged(int)</signal>
    <signal>yRotationChanged(int)</signal>
    <signal>zRotationChanged(int)</signal>
    <signal>zoomChanged(double)</signal>
    <signal>translateStepChanged(double)</signal>
    <signal>xTranslationChanged(double)</signal>
    <signal>yTranslationChanged(double)</signal>
    <signal>zTranslationChanged(double)</signal>
    <slot>setXRotation(int)</slot>
    <slot>setYRotation(int)</slot>
    <slot>setZRotation(int)</slot>
    <slot>setZoom(double)</slot>
    <slot>setMouseMode(int)</slot>
    <slot>setShowMesh(bool)</slot>
    <slot>setShowCube(bool)</slot>
    <slot>setTranslateStep(double)</slot>
    <slot>setVisualisationMode(int)</slot>
    <slot>setColormapName(int)</slot>
    <slot>setXTranslation(double)</slot>
    <slot>setYTranslation(double)</slot>
    <slot>setZTranslation(double)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>spinX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>214</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>214</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinZ</sender>
   <signal>valueChanged(int)</signal>
   <receiver>sliderZ</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>199</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderZ</sender>
   <signal>valueChanged(int)</signal>
   <receiver>spinZ</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>174</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>199</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>xRotationChanged(int)</signal>
   <receiver>sliderX</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>251</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>527</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderX</sender>
   <signal>valueChanged(int)</signal>
   <receiver>picture</receiver>
   <slot>setXRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>527</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>280</x>
     <y>174</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>yRotationChanged(int)</signal>
   <receiver>sliderY</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>285</x>
     <y>80</y>
    </hint>
    <hint type="destinationlabel">
     <x>595</x>
     <y>190</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderY</sender>
   <signal>valueChanged(int)</signal>
   <receiver>picture</receiver>
   <slot>setYRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>595</x>
     <y>190</y>
    </hint>
    <hint type="destinationlabel">
     <x>262</x>
     <y>146</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>zRotationChanged(int)</signal>
   <receiver>sliderZ</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>283</x>
     <y>212</y>
    </hint>
    <hint type="destinationlabel">
     <x>662</x>
     <y>167</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sliderZ</sender>
   <signal>valueChanged(int)</signal>
   <receiver>picture</receiver>
   <slot>setZRotation(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>662</x>
     <y>139</y>
    </hint>
    <hint type="destinationlabel">
     <x>214</x>
     <y>220</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>spinZoom</sender>
   <signal>valueChanged(double)</signal>
   <receiver>picture</receiver>
   <slot>setZoom(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>664</x>
     <y>234</y>
    </hint>
    <hint type="destinationlabel">
     <x>331</x>
     <y>317</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>zoomChanged(double)</signal>
   <receiver>spinZoom</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>300</x>
     <y>275</y>
    </hint>
    <hint type="destinationlabel">
     <x>664</x>
     <y>234</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>mouseMode</sender>
   <signal>activated(int)</signal>
   <receiver>picture</receiver>
   <slot>setMouseMode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>665</x>
     <y>283</y>
    </hint>
    <hint type="destinationlabel">
     <x>367</x>
     <y>359</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showMesh</sender>
   <signal>clicked(bool)</signal>
   <receiver>picture</receiver>
   <slot>setShowMesh(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>469</y>
    </hint>
    <hint type="destinationlabel">
     <x>268</x>
     <y>335</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>showRegionBorder</sender>
   <signal>clicked(bool)</signal>
   <receiver>picture</receiver>
   <slot>setShowCube(bool)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>666</x>
     <y>490</y>
    </hint>
    <hint type="destinationlabel">
     <x>254</x>
     <y>352</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>translateStep</sender>
   <signal>valueChanged(double)</signal>
   <receiver>picture</receiver>
   <slot>setTranslateStep(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>665</x>
     <y>316</y>
    </hint>
    <hint type="destinationlabel">
     <x>443</x>
     <y>394</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>translateStepChanged(double)</signal>
   <receiver>translateStep</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>449</x>
     <y>255</y>
    </hint>
    <hint type="destinationlabel">
     <x>665</x>
     <y>316</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>visualizationMode</sender>
   <signal>activated(int)</signal>
   <receiver>picture</receiver>
   <slot>setVisualisationMode(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>560</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>315</x>
     <y>372</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>colorMap</sender>
   <signal>activated(int)</signal>
   <receiver>picture</receiver>
   <slot>setColormapName(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>664</x>
     <y>446</y>
    </hint>
    <hint type="destinationlabel">
     <x>417</x>
     <y>416</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xTranslation</sender>
   <signal>valueChanged(double)</signal>
   <receiver>picture</receiver>
   <slot>setXTranslation(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>566</x>
     <y>331</y>
    </hint>
    <hint type="destinationlabel">
     <x>313</x>
     <y>434</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yTranslation</sender>
   <signal>valueChanged(double)</signal>
   <receiver>picture</receiver>
   <slot>setYTranslation(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>571</x>
     <y>360</y>
    </hint>
    <hint type="destinationlabel">
     <x>302</x>
     <y>420</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>zTranslation</sender>
   <signal>valueChanged(double)</signal>
   <receiver>picture</receiver>
   <slot>setZTranslation(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>587</x>
     <y>386</y>
    </hint>
    <hint type="destinationlabel">
     <x>380</x>
     <y>454</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>xTranslationChanged(double)</signal>
   <receiver>xTranslation</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>275</x>
     <y>439</y>
    </hint>
    <hint type="destinationlabel">
     <x>598</x>
     <y>328</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>yTranslationChanged(double)</signal>
   <receiver>yTranslation</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>408</x>
     <y>475</y>
    </hint>
    <hint type="destinationlabel">
     <x>632</x>
     <y>360</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>picture</sender>
   <signal>zTranslationChanged(double)</signal>
   <receiver>zTranslation</receiver>
   <slot>setValue(double)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>145</x>
     <y>352</y>
    </hint>
    <hint type="destinationlabel">
     <x>614</x>
     <y>388</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
